### PR TITLE
Add code examples for libstd/time

### LINKED ISSUE #29379: API Docs: time
Part of https://github.com/rust-lang/rust/issues/29329

http://doc.rust-lang.org/std/time/

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #29379.

r? @steveklabnik

### CODE REVIEW

File: src/libstd/time/mod.rs [hunk 1]
```diff
@@ -9,6 +9,14 @@
 // except according to those terms.
 
 //! Temporal quantification.
+//!
+//! Example:
+//!
+//! ```
+//! use std::time::Duration;
+//!
+//! let five_seconds = Duration::new(5, 0);
```
Comment (nodakai) [inline src/libstd/time/mod.rs]: I think [`Duration::from_secs()`](https://doc.rust-lang.org/nightly/std/time/struct.Duration.html#method.from_secs) is handy here.
You may also check `assert_eq!(Duration::new(5, 0), Duration::from_secs(5))`

File: src/libstd/time/mod.rs [hunk 2]
```diff
@@ -40,6 +48,22 @@ mod duration;
...
+///
+/// ```
+/// use std::time::{Duration, Instant};
+/// use std::thread::sleep;
+///
+/// fn main() {
+///    let now = Instant::now();
+///
+///    // we sleep for 2 seconds
+///    sleep(Duration::new(2, 0));
```
Comment (nodakai) [inline src/libstd/time/mod.rs]: This piece of code is actually executed by `make check-docs`. I wonder what might be the impact of sleeping here.

Comment (GuillaumeGomez) [inline src/libstd/time/mod.rs]: Well thought! I'll add an ignore instruction.

File: src/libstd/time/mod.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/libstd/time/mod.rs]: Well saw, I'll add this line.

**Pull request merged**